docs/examples/snek5000-tgv/src/snek5000_tgv/tgv.usr.f

```diff
@@ -80,6 +80,14 @@ subroutine userchk()
 
       character*80 fnames(3)
 
+      if (istep.eq.0) then
+         if (nid.eq.0) then
+            open(10, File='spatial_means.txt', position='append')
+            write(10,*) 'Time  ', 'Energy  ', 'Enstrophy  '
+            close(10)
```

Comment:
Check for file existence before writing the header.

Reply:
Otherwise restarts may not work.

Reply:
I don't think it is very bad if the header is written more than once. What if important is that the header starts with `"#"`.

Reply:
It is very simple to do.

https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vna3/index.html

Reply:
I think it can be useful for use to have a line in this file for each new session. So you can really keep this write.

Reply:
I disagree. I know with pandas it is easy to ignore such lines, but not everyone would be postprocessing using the same tools (I am thinking gnuplot, Octave, Fortran or even a spreadsheet)

While creating data, it is important to do things well, if it is not too much work. Especially in publically shared code.

Reply:
I have no strong feelings about this but I strongly suspect that a comment with a line starting with `#` cannot be a problem for a tool made to read csv files. I check for few tools (in R, Matlab, Octave, Fortran, Java, gnuplot), and there are very easy solutions to skip lines starting with `#`.

Reply:
OK, then we leave it as it is now then, with header written once. I am OK with a `#` in the beginning.